### PR TITLE
MM-10164 Add e2e on extra metadata when clicking interactive button

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Add E2E to verify extra metadata when clicking interactive button

#### Ticket Link
Jira ticket: https://mattermost.atlassian.net/browse/MM-10164

### CODE REVIEW

File: e2e/cypress/fixtures/hooks/message_buttons.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+    "attachments": [
+        {
+            "pretext": "This is the attachment pretext.",
+            "text": "This is the attachment text for Interactive Message Buttons.",
+            "actions": [
+                {
+                    "name": "Ephemeral Message",
```
Comment (hanzei) [inline e2e/cypress/fixtures/hooks/message_buttons.json]: Maybe add 
```suggestion
                    "name": "Ephemeral Message",
                    "type": "button",
```

To be explicit?

Comment (hanzei) [reply e2e/cypress/fixtures/hooks/message_buttons.json]: Or even better, have both a test with and without this being set and confirm the behaviour is the same.

File: e2e/cypress/integration/interactive_button/interactive_button_spec.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            cy.findByText('channel_name: "town-square"').should('be.visible');
+            cy.apiGetChannelByName('ad-1', 'town-square').then((channel) => {
+                cy.findByText(`channel_id: "${channel.id}"`).should('be.visible');
+                cy.findByText('channel_name: "town-square"').should('be.visible');
+                cy.findByText(`team_id: "${channel.team_id}"`).should('be.visible');
+                cy.findByText('team_domain: "ad-1"').should('be.visible');
+            });
+            cy.findByText(`post_id: "${interactiveButtonPostId}"`).should('be.visible');
+            cy.findByText(/trigger_id:/).should('be.visible');
+            cy.findByText('type: ""').should('be.visible');
```
Comment (hanzei) [inline e2e/cypress/integration/interactive_button/interactive_button_spec.js]: Hm, I'm surprised by this. The expected behaviour from the top of my head should be that `button` is returned, even if it's not set.

If you agree, would you open a bug ticket to fix this on the server?

File: e2e/cypress/utils/index.js [hunk 3]
```diff
@@ -47,6 +48,15 @@ export function getMessageMenusPayload({dataSource, options} = {}) {
     return data;
 }
 
+export function getMessageButtonsPayload() {
+    const data = messageButtons;
+    const callbackUrl = Cypress.env().webhookBaseUrl + '/message_buttons';
+    data.attachments[0].actions[0].integration.url = callbackUrl;
+    data.attachments[0].actions[1].integration.url = callbackUrl;
```
Comment (hanzei) [inline e2e/cypress/utils/index.js]: I would like to see this function to not use fixed indices but instead set the url for all attachments and actions.

File: e2e/cypress/integration/interactive_button/interactive_button_spec.js (hunk 2 above)
Comment (saturninoabril) [reply e2e/cypress/integration/interactive_button/interactive_button_spec.js]: Sure, will check that.

File: e2e/cypress/utils/index.js (hunk 3 above)
Comment (saturninoabril) [reply e2e/cypress/utils/index.js]: Sure, though it's just a good to have but not necessary since it's based on fixed and not dynamic JSON values.

Comment (hanzei) [reply e2e/cypress/utils/index.js]: I'm 1/5 that someone will stumble across this at some point and will not set the URL correctly.

But not a blocker.

File: e2e/cypress/integration/interactive_button/interactive_button_spec.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            cy.get(`#messageAttachmentList_${postId}`).as('messageAttachmentList');
+        });
+
+        // * Verify each element of message attachment list and then click "Ephemeral" button
+        cy.get('@messageAttachmentList').within(() => {
+            cy.get('.attachment__thumb-pretext').should('be.visible').and('have.text', 'This is the attachment pretext.');
+            cy.get('.post-message__text-container').should('be.visible').and('have.text', 'This is the attachment text for Interactive Message Buttons.');
+            cy.findByText('Ephemeral Message').should('be.visible').click();
+        }).then(() => {
+            // * Verify metadata on click to "Ephemeral" button
```
Comment (prapti) [inline e2e/cypress/integration/interactive_button/interactive_button_spec.js]: Is clicking the "Ephemeral Message" button supposed to do anything? Right now, nothing happens when the button is clicked, and the test for me is failing right after step in verifyMessage (the last post message cannot be found so I'm wondering if this was the message that the Ephemeral Message button was supposed to generate):
`CypressError: Timed out retrying: expected '<div#postMessageText_d4k857ue5pyemfyzfxr6dmr3we.post-message__text>' to be 'visible'`

**Comment by saturninoabril** [comment]
@hanzei @prapti I didn't got a chance to revisit this PR due to some priority.  Will get back into this later this week, thanks for the reviews.

**Comment by saturninoabril** [comment]
Closing this.

**Pull request closed without merging**